packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -134,6 +146,88 @@ function prepCompilerMaps<
   return { objMap, entryMap };
 }
 
+/**
+ * Helper function for the tuple compilation phase for prove or verify.
+ * Input information is gathered into a map for easy lookup by name when
+ * compiling data that depends on tuples.
+ *
+ * The tuples are sorted by name before they are processed.
+ *
+ * @param config proof config
+ * @param entryMap map for looking up entries by identifier
+ * @param paramMaxEntries maximum number of entries allowed by the chosen
+ * circuit description.
+ * @param paramMaxTuples maximum number of tuples allowed by the chosen
+ * circuit description.
+ * @param paramTupleArity tuple arity used by the chosen circuit description.
+ * @param inputs input object for prove or verify
+ *   (GPCProofInput or GPCRevealedClaims).
+ * @returns maps for looking up objects by name, and entries by identifier.
+ *   See the map value types for what's included.
+ */
+function prepCompilerTupleMap<ObjInput extends POD | GPCRevealedObjectClaims>(
+  config: GPCBoundConfig,
+  entryMap: Map<PODEntryIdentifier, CompilerEntryInfo<ObjInput>>,
+  paramMaxEntries: number,
+  paramMaxTuples: number,
+  paramTupleArity: number
+): Map<TupleIdentifier, CompilerTupleInfo> {
+  // And now for the tuple map, which is sorted in alphabetical order and
+  // populated with the corresponding values from entryMap. Note that the index
+  // here will be offset by the maximum number of entries in the circuit itself.
+  const tupleMap = new Map();
+  if (config.tuples !== undefined) {
+    let tupleIndex = paramMaxEntries;
+    const tupleNameOrder: PODName[] = Object.keys(
+      config.tuples
+    ).sort() as PODName[];
+    for (const tupleName of tupleNameOrder) {
+      const tupleConfig = config.tuples[tupleName];
+      if (tupleConfig === undefined) {
+        throw new Error(`Missing config for tuple ${tupleName}.`);
+      }
+
+      if (tupleConfig.entries.length < 2) {
+        throw new Error(`Arity of tuple ${tupleName} is less than 2.`);
+      }
+
+      const tupleEntryRef: number[] = tupleConfig.entries.map((entryId) => {
+        const entryIdx = entryMap.get(entryId)?.entryIndex;
+
+        if (entryIdx === undefined) {
+          throw new ReferenceError(
+            `Missing entry index for identifier ${entryId} in tuple ${tupleName}.`
+          );
+        }
+
+        return entryIdx;
+      });
+
+      // Encode tuple as sequence of tuples of arity paramTupleArity
+      const tupleIndices = computeTupleIndices(
+        paramTupleArity,
+        tupleIndex,
+        tupleEntryRef
+      );
+
+      tupleIndex += tupleIndices.length;
+
+      tupleMap.set(`tuple.${tupleName}`, {
+        tupleRef: tupleIndex - 1,
```

Comment:
This needs an explanation.  I think I puzzled out that this is the index of the last tuple in the chain, which is the one with the relevant value.  That needs documentation. Also I think I'd call that `tupleIndex` since it's used in the same way as `entryIndex` and `objIndex` elsewhere.

Reply:
I had reservations about using `tupleIndices` and `tupleIndex` in the same context when they have slightly different meanings, hence `tupleRef`. I'll make the change.

Reply:
Maybe rename `tupleIndices` to `tupleInputIndices`?